cognite/client/utils/_time.py

```diff
@@ -556,3 +557,39 @@ def to_pandas_freq(granularity: str, start: datetime) -> str:
         }[floored.month]
 
     return f"{multiplier}{unit}"
+
+
+def _unit_in_days(unit: str, ceil: bool = True) -> float:
+    if unit in {"w", "d", "h", "m", "s"}:
+        unit = GRANULARITY_IN_TIMEDELTA_UNIT[unit]
+        arg = {unit: 1}
+        return timedelta(**arg) / timedelta(hours=1)
+
+    if unit == "month":
+        days = 31.0 if ceil else 28.0
+    elif unit == "quarter":
+        days = 92.0 if ceil else 91.0
+    else:  # years
+        days = 366.0 if ceil else 365.0
+    return days
```

Comment:
Is this only used for approximation on when to raise? (Since `month` typically is more likely to be 30 than 28)

If so (_should not be used for precise calculations_), let's state that explicitly in the docstring (or function name maybe)

----

Last option, just hardcode the max upper limits for week, month, quarter and year? 🤔 🤷

Reply:
Made it private to avoid it being used, but will add that to the docstring as well